client/memconfig.go

```diff
@@ -0,0 +1,153 @@
+		PutReturnsFailoverError: c.PutReturnsFailoverError,
+	})
+}
+
+
+// intermediaryCfg ... used for decoding into a less rich type before
+// translating to a structured MemConfig
+type intermediaryCfg struct {
+	MaxBlobSizeBytes        uint64
+	BlobExpiration          string
+	PutLatency              string
+	GetLatency              string
+	PutReturnsFailoverError bool
+}
+
+// IntoMemConfig ... converts an intermediary config into a memconfig 
+// with structured type definitions
+func (cfg *intermediaryCfg) IntoMemConfig() (*MemConfig, error) {
+	getLatency, err := time.ParseDuration(cfg.GetLatency)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse getLatency: %w", err)
+	}
+
+	putLatency, err := time.ParseDuration(cfg.PutLatency)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse putLatency: %w", err)
+	}
+
+	blobExpiration, err := time.ParseDuration(cfg.BlobExpiration)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse blobExpiration: %w", err)
+	}
+
+	return &MemConfig{
+		MaxBlobSizeBytes: cfg.MaxBlobSizeBytes,
+		BlobExpiration: blobExpiration,
+		PutLatency: putLatency,
+		GetLatency: getLatency,
+		PutReturnsFailoverError: cfg.PutReturnsFailoverError,
+	}, nil
+}
+
+
+// MemCfgClient implements a standard client for the eigenda-proxy
+// that can be used for updating a memstore configuration in real-time
+// this is useful for API driven tests in protocol forks that leverage
+// custom integrations with EigenDA
+type MemCfgClient struct {
+	cfg        *Config
+	httpClient *http.Client
+}
+
+// NewMemCfgClient ... client constructor
+func NewMemCfgClient(cfg *Config) *MemCfgClient {
+	cfg.URL = cfg.URL + memConfigEndpoint // initialize once to avoid unnecessary ops when patch/get
+
+	scc := &MemCfgClient{
+		cfg: cfg,
+		httpClient: http.DefaultClient,
+	}
+
+	return scc
+}
+
+// decodeResponseToMemCfg ... converts http response to structured MemConfig
+func decodeResponseToMemCfg(resp *http.Response) (*MemConfig, error) {
+	var cfg intermediaryCfg
+	if err := json.NewDecoder(resp.Body).Decode(&cfg); err != nil {
+		return nil, err
+	}
+	return cfg.IntoMemConfig()
+}
+// GetConfig retrieves the current configuration.
+func (c *MemCfgClient) GetConfig(ctx context.Context) (*MemConfig, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.cfg.URL, &bytes.Buffer{})
+	if err != nil {
+		return nil, err 
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to read config. expected status code 200, got: %d", resp.StatusCode)
+	}
+
+	return decodeResponseToMemCfg(resp)
+}
+
+// UpdateConfig updates the configuration using the new MemConfig instance
+// by sending a PATCH request to the proxy server
+func (c *MemCfgClient) UpdateConfig(ctx context.Context, update *MemConfig) (*MemConfig, error) {
+	body, err := update.MarshalJSON()
```

Comment:
This is a bit dangerous the way its implemented, because it doesn't really allow you to PATCH. You should prob use `omitempty` tag in the struct. Otherwise any field that is not initialized will be given zero value and happily overwrite the server's config with those values. We want to make sure zero fields are not included in the marshalled json (except for boolean values...!)

Reply:
hmmm how would fields be omitted if we're passing in a structured `MemConfig` to update? This would make more sense if we providing an arbitrary key/value interface which we're not. This client functionality should work fine for the use case of Arbitrum E2E testing and shouldn't result in fields being discarded.

Reply:
The problem is that because it's a PATCH, someone could pass in
```
{
	PutReturnsFailoverError: true
}
```
thinking they'll just update this single value, but actually what this will do is set all the other fields to their zero value.

This is why I'm suggesting instead of this method taking in a *MemConfig, it instead takes a [ConfigUpdate](https://github.com/Layr-Labs/eigenda-proxy/blob/d7b1642956de39bfc62b1bfe3a4467fe4e67bc9a/store/generated_key/memstore/memconfig/http_handlers.go#L14) like I used in the proxy side.

Reply:
There are 2 other options that you might prefer:
1. we forget that I implemented this as a PATCH, and we add a big disclaimer in the comment to this method saying that this method is actually implemented as a PUT, where the EXACT MemConfig passed as input will be set on the proxy. We then suggest that a client gets the current MemConfig first, and then make modifications to that before calling UpdateConfig.
2. We define individual setters for each field which will create a patch with only that specific field. This might be cleaner actually.. with the downside that if you need to update multiple fields at once you'll need to make N calls.

Reply:
would prefer to just keep it as is. currently we only plan to use this in the arbitrum integration for testing failover. reading the entire config to update a specific field should be ok for the existing use case.

Reply:
Fine with me, although I PROMISE you that this will cause a bug at some point.
Can you at least add the disclaimer comment I suggested in option 1 above then?